nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/query/NxQueryBuilder.java

```diff
@@ -281,22 +282,25 @@ protected boolean nxqlHasSelectClause(String nxql) {
 
     protected String guessFieldType(String field) {
         String fieldType;
+        if (ES_SCORE_FIELD.equals(field)) {
+            // this special field should not have an unmappedType
+            return null;
+        }
         try {
             SchemaManager schemaManager = Framework.getService(SchemaManager.class);
             fieldType = schemaManager.getField(field).getType().getName();
         } catch (NullPointerException e) {
             // probably an internal field without schema
-            fieldType = "string";
+            fieldType = "keyword";
```

Comment:
Is this strictly related to the RestClient switch? I would prefer a separate commit for all unrelated cleanups...